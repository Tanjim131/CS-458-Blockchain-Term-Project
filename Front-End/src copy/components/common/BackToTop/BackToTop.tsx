import { Affix, BackTop, Col, Row, Tooltip } from "antd";
import React, { FC } from "react";
import Icon_Back_Top from "../../../../public/svg/icon_back_to_top.svg";

interface IBackToTopProps {}

/**
 * Back to Top component.
 *
 * @Usage: <BackToTop />
 */
const BackToTop: FC<IBackToTopProps> = (props: IBackToTopProps) => {
  return (
    <BackTop>
      <Tooltip placement="top" title={"Back to Top"}>
        <img src={Icon_Back_Top} />
      </Tooltip>
    </BackTop>
  );
};

export default BackToTop;
