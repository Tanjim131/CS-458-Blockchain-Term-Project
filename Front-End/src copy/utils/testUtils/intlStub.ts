import { createIntl } from "react-intl";
const en_us = require("../../i18n/translations/en.json");

export const intlStub = createIntl({ locale: "en-us", messages: en_us });
export const intlMessages = en_us;
