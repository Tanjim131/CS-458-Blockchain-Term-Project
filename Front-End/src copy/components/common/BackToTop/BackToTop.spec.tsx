import React from 'react';
import { shallow } from "enzyme";
import BackToTop from "./BackToTop";
import { shallowToJson } from "enzyme-to-json";
import { intlStub } from '../../../../utils/testUtils/intlStub';

describe('BackToTop Component', () => {
    let wrapper: any;

    beforeAll(() => {
        wrapper = shallow(
            <BackToTop
            />
        )
    })

    it('Check if component exists or not and capture snapshot', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })

})