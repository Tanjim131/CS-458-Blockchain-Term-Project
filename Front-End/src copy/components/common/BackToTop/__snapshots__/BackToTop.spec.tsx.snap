// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackToTop Component Check if component exists or not and capture snapshot 1`] = `
<Memo(BackTop)>
  <Tooltip
    arrowPointAtCenter={false}
    autoAdjustOverflow={true}
    mouseEnterDelay={0.1}
    mouseLeaveDelay={0.1}
    placement="top"
    title="Back to Top"
    transitionName="zoom-big-fast"
  >
    <img
      src="mock-file"
    />
  </Tooltip>
</Memo(BackTop)>
`;
